### PR TITLE
[IMP] open_academy: Add program to test web services T#59013

### PR DESCRIPTION
**Pull request by geraspace**
- Display sessions, and their corresponding number of seats.
- Create a new session for one of the courses.

- Fix start and end dates in session model.

### CODE REVIEW

**Comment by geraspace** [comment]
Good morning, @edy1192. Here's the web service PR. Thank you!

File: open_academy/web_services/ws.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+print("Available instructors:")
+for instructor in instructors_info:
+    print(instructor["name"])
+
+# Fill the information
+course_name = "Nikon"
+new_session_name = "Nikon II"
+duration_in_days = 1
+number_of_seats = 5
+instructor_name = "Brandon Freeman"
```
Comment (edy1192) [inline open_academy/web_services/ws.py]: You could create a dictionary with this 

```suggestion
session_data = {
    "name": "Nikon II",
    "course_id": course_id,
    "instructor_id": instructor_id,
    "duration": 1,
    "seats": 5,
}
```

File: open_academy/web_services/ws.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+id = models.execute_kw(db, uid, password, 'open.academy.session', 'create', [{
+    "name": new_session_name,
+    "course_id": course_id,
+    "duration": duration_in_days,
+    "seats": number_of_seats,
+    "instructor_id": instructor_id,
+    "attendee_ids": [7, 3]
+    }]
+)
```
Comment (edy1192) [inline open_academy/web_services/ws.py]: Use the dictionary `session_data`.
```suggestion
id = models.execute_kw(db, uid, password, 'open.academy.session', 'create', session_data)
```

File: open_academy/web_services/ws.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+course_id = (next(filter(lambda d: d.get("name") == course_name, courses_info), None))["id"]
+instructor_id = (next(filter(lambda d: d.get("name") == instructor_name, instructors_info), None))["id"]
+
+id = models.execute_kw(db, uid, password, 'open.academy.session', 'create', [{
+    "name": new_session_name,
+    "course_id": course_id,
+    "duration": duration_in_days,
+    "seats": number_of_seats,
+    "instructor_id": instructor_id,
+    "attendee_ids": [7, 3]
```
Comment (edy1192) [inline open_academy/web_services/ws.py]: Never add harcoded ID's, please. I learned it after my first review.

Read about the special commands to update o2m and m2m fields [here](https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html#odoo.models.Model.write)

Then use that to add the attendee_id in the dictionary.

File: open_academy/web_services/ws.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# Create a session
+
+courses_info = models.execute_kw(db, uid, password, 'open.academy.course',
+                                 'search_read', [], {'fields': ['name']})
+instructors_info = models.execute_kw(db, uid, password, 'res.partner',
+                                     'search_read', [[['instructor', '=', True]]],
+                                     {'fields': ['name']})
+possible_attendees = models.execute_kw(db, uid, password, 'res.partner',
+                                 'search_read', [], {'fields': ['name']})
```
Comment (edy1192) [inline open_academy/web_services/ws.py]: You just need one course and one instructor.
```suggestion
course_id = models.execute_kw(db, uid, password, 'open.academy.course',
                                 'search', [], {'limit': 1})
instructor_id = models.execute_kw(db, uid, password, 'res.partner',
                                     'search', [[['instructor', '=', True]]],
                                     {'limit': 1})
attendee_id = models.execute_kw(db, uid, password, 'res.partner',
                                 'search', [], {'limit': 1})
```

File: open_academy/web_services/ws.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+for attendee in possible_attendees:
+    print(attendee["id"], attendee["name"])
+print("\n")
+print("Available courses:")
+for course in courses_info:
+    print(course["name"])
+print("\n")
+print("Available instructors:")
+for instructor in instructors_info:
+    print(instructor["name"])
```
Comment (edy1192) [inline open_academy/web_services/ws.py]: With the search and limit 1 you don't need this.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
@edy1192 I think it is working now :) Thanks!

**Review by edy1192 — APPROVED** [review]

**Pull request merged**